engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/sink/flink/FlinkKafkaSinkWithCacheableProducer.scala

```diff
@@ -0,0 +1,71 @@
+package pl.touk.nussknacker.engine.kafka.sink.flink
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.java.ClosureCleaner
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.runtime.state.{FunctionInitializationContext, FunctionSnapshotContext}
+import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction
+import org.apache.flink.streaming.api.functions.sink.{RichSinkFunction, SinkFunction}
+import org.apache.flink.streaming.connectors.kafka.{FlinkKafkaErrorCode, FlinkKafkaException, KafkaSerializationSchema}
+import pl.touk.nussknacker.engine.api.MetaData
+import pl.touk.nussknacker.engine.kafka.KafkaProducerCreator
+import pl.touk.nussknacker.engine.kafka.sharedproducer.{LoggingSharedKafkaProducer, LoggingSharedKafkaProducerHolder}
+import pl.touk.nussknacker.engine.util.SynchronousExecutionContext
+
+import scala.concurrent.ExecutionContext
+
+class FlinkKafkaSinkWithCacheableProducer[IN](kafkaProducerCreator: KafkaProducerCreator.Binary,
```

Comment:
1. Why do you need to cache producers in sinks? Caching was for purpose when there was many usages of kafka producers in the same scenario (mainly for exception handling). For normal (sink) usage, you don't have many sinks in the scenario.
2. How to use it? E.g. with our universal kafka sink.